### PR TITLE
47970 add elemmatch operator

### PR DESCRIPTION
**Pull request by mrix**
Added new `elemMatch()` operator

### CODE REVIEW

**Comment by hairmare** [comment]
I don't see a lot of symfony specific code in this PR. Surely most of this could be added to https://github.com/libgraviton/php-rql-parser just as well.

**Comment by mrix** [comment]
@hairmare I just thought https://github.com/libgraviton/php-rql-parser is plain RQL -> MongoDb converter. And this repository is Graviton specific code.
I can move these changes to https://github.com/libgraviton/graviton/pull/311

**Comment by hairmare** [comment]
@mrix You are right. php-rql-parser is plain RQL -> MongoDB. The bundle is here to integrate that with symfony/graviton.

What part of this is graviton specific?

**Comment by mrix** [comment]
@hairmare I have a task about it in our issue tracker. I will ask Daimen about EVO task number.
No problem. I can move these changes to graviton repository.

**Comment by hairmare** [comment]
I think they should go to  https://github.com/libgraviton/php-rql-parser.

**Comment by hairmare** [comment]
Not graviton in any case!

**Comment by mrix** [comment]
Ok

**Comment by mrix** [comment]
@hairmare I have reopened this PR because I need to change RQL parser class.

**Comment by hairmare** [comment]
good to merge as soon as `composer.json` has been updated to pull in the `v2.0.0-alpha15` tag of `graviton/php-rql-parser`.

**Comment by mrix** [comment]
@hairmare done

**Comment by hairmare** [comment]
Please add the requirement to `composer.json` as well. This will break if anyone doesn't update them both.

File: composer.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
     "require": {
         "php": "~5.4",
         "symfony/http-kernel": "~2.6",
-        "graviton/php-rql-parser": "~2.0@alpha"
+        "graviton/php-rql-parser": "~2.0@alpha,>=2.0.0-alpha15"
```
Comment (mrix) [inline composer.json]: @hairmare Is it correct?

Comment (hairmare) [inline composer.json]: the `~2.0@alpha,` part can be removed

**Comment by hairmare** [comment]
tagged as `v0.10.0` after updating the `composer.lock` file.

**Pull request merged**